reviewrot/githubstack.py

```diff
@@ -0,0 +1,166 @@
+                             username or organization
+            state_ (str): The filter(state) for pull requests, e.g, older
+                        or newer
+            value (int): The value in terms of duration for requests
+                         to be older or newer than
+            duration (str): The duration in terms of period(year,
+                            month, hour, minute) for requests to be
+                            older or newer than.
+        """
+        try:
+            # get repository object for given user/organization and repo name
+            repo = uname.get_repo(repo_name)
+        except UnknownObjectException as e:
+            log.debug('Repository %s not found for user %s',
+                      repo_name, uname.login)
+            log.debug(e)
+            raise Exception('Repository %s not found for user %s'
+                            % (repo_name, uname.login))
+        log.debug('Looking for pull requests for %s -> %s/%s ',
+                  'github', uname.login, repo_name)
+        # get list of open pull requests for a given repository
+        pull_requests = repo.get_pulls()
+        if len(list(pull_requests)) == 0:
+            log.debug('No open pull requests found for %s/%s ',
+                      uname.login, repo_name)
+        res_ = []
+        for pr in pull_requests:
+            """
+            find the relative time difference between now and
+            pull request filed to retrieve relative information
+            """
+            rel_diff = relativedelta(datetime.datetime.now(),
+                                     pr.created_at)
+            """
+            find the absolute time difference between now and
+            pull request filed to retrieve absolute information
+            """
+            abs_diff = datetime.datetime.now() - pr.created_at
+            if (state_ is not None and value is not None and
+                    duration is not None):
+                """
+                check if pull request is older/newer than specified time
+                interval
+                """
+                result = self.check_request_state(abs_diff, rel_diff, state_,
+                                                  value, duration)
+                # skip the request if it doesn't match the specified criteria
+                if not result:
+                    log.debug("pull request '%s' is not %s than specified"
+                              " time interval", pr.title, state_)
+                    continue
+            # format the time interval pull request has been filed since
+            time = self.format_duration(rel_diff)
+            # fetch and format comments for pull request
+            comments = []
+            if(pr.comments == 1):
+                comments.append('%s' % ', with 1 comment')
+            elif(pr.comments > 1):
+                comments.append('%s %s %s' % (', with', str(pr.comments),
```

Comment:
Minor thing: no need to use `str` to convert `pr.comments`. This is automatically done with the string formatting.